src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -85,5 +133,212 @@ public override bool Execute()
+                {
+                    continue;
+                }
+
+                string[] packages = Directory.GetFiles(path, "*.nupkg", SearchOption.AllDirectories);
+                foreach (string package in packages)
+                {
+                    NupkgInfo info = GetNupkgInfo(package);
+                    string id = info.Id.ToLower();
+                    string version = info.Version.ToLower();
+
+                    // Add package with version to appropriate hashtable
+                    if (packageSource.StartsWith("source-built-"))
+                    {
+                        AddToDictionary(currentPackages, id, version);
+                    }
+                    else if (packageSource.Equals("reference-packages"))
+                    {
+                        AddToDictionary(referencePackages, id, version);
+                    }
+                    else // previously built packages
+                    {
+                        AddToDictionary(previouslyBuiltPackages, id, version);
+                    }
+
+                    AddToDictionary(allSourcesPackages, packageSource, id);
+                }
+            }
+        }
+
+        private void DiscoverPackagesFromSbrpCacheSource(XElement pkgSourcesElement, Dictionary<string, List<string>> allSourcesPackages, Dictionary<string, List<string>> currentPackages)
+        {
+            // 'source-build-reference-package-cache' is a dynamic source, populated by SBRP build.
+            // Discover all SBRP packages from checked in nuspec files.
+
+            if (!Directory.Exists(SbrpRepoSrcPath))
+            {
+                throw new InvalidDataException(string.Format(CultureInfo.CurrentCulture, "SBRP repo root does not exist in expected path: {0}", SbrpRepoSrcPath));
+            }
+
+            string[] nuspecFiles = Directory.GetFiles(SbrpRepoSrcPath, "*.nuspec", SearchOption.AllDirectories);
+            foreach (string nuspecFile in nuspecFiles)
+            {
+                try
+                {
+                    using Stream stream = File.OpenRead(nuspecFile);
+                    NupkgInfo info = GetNupkgInfo(stream);
+                    string id = info.Id.ToLower();
+                    string version = info.Version.ToLower();
+
+                    AddToDictionary(currentPackages, id, version);
+                    AddToDictionary(allSourcesPackages, SbrpCacheSourceName, id);
+                }
+                catch (Exception ex)
+                {
+                    throw new InvalidDataException(string.Format(CultureInfo.CurrentCulture, "Invalid nuspec file", nuspecFile), ex);
+                }
+            }
+        }
+
+        private void GetExistingFilteredSourceMappings(XElement pkgSrcMappingElement, Dictionary<string, List<string>> currentPackages, Dictionary<string, List<string>> referencePackages, Dictionary<string, List<string>> previouslyBuiltPackages, Dictionary<string, List<string>> oldSourceMappingPatterns)
```

Comment:
This line is too long IMO.
```suggestion
        private void GetExistingFilteredSourceMappings(
            XElement pkgSrcMappingElement,
            Dictionary<string, List<string>> currentPackages,
            Dictionary<string, List<string>> referencePackages,
            Dictionary<string, List<string>> previouslyBuiltPackages,
            Dictionary<string, List<string>> oldSourceMappingPatterns)
```

Reply:
Fixed with https://github.com/dotnet/installer/pull/19114/commits/3f458e5268b5925b8fae42f35138b3f344b24f28